src/sdsc/__init__.py

```diff
@@ -117,6 +114,30 @@ def sanitizepunctuation(text, quotes=False, apostrophes=False):
         text = apostrophe.sub('\'', text)
     return text
 
+def findtagreplacement(text):
+    """Search through a token to see whether it contains a replaced tag.
+    Allows finding a single tag replacement only.
+
+    text - str(): a token
+    """
+
+    tagreplacement = re_compile('##@(\w+)(?:-(\d+))?##')
+
+    tagsreplaced = tagreplacement.search(text)
+
+    if tagsreplaced:
+        tagtype = str(tagsreplaced.group(1))
+        try:
+            # This group is optional (though it should now always be generated)
```

Comment:
If it *should* always be generated, not being there is a bug and should probably be reported.